### PR TITLE
Create GH Action to reliably determine mergeability

### PR DESCRIPTION
**Pull request by roryabraham**
### Details
When getting ready to automerge a PR, we want to first make sure that it's mergeable. However, Github uses a background job to calculate mergeability... from the GH docs:

> The value of the mergeable attribute can be true, false, or null. If the value is null, then GitHub has started a background job to compute the mergeability. After giving the job time to complete, resubmit the request. When the job finishes, you will see a non-null value for the mergeable attribute in the response. If mergeable is true, then merge_commit_sha will be the SHA of the test merge commit

So we can't rely on the boolean result of the mergeable attribute to be readily available. This PR creates a custom Github Action to continually poll mergeability of a PR until the result is non-null, and output the end result.

### Fixed Issues
Fixes failed workflow run: https://github.com/Expensify/Expensify.cash/runs/2133374703?check_suite_focus=true

### Tests
Added automated tests for the new JS action.

Need to test it's usage in the workflow by merging this PR.

### CODE REVIEW

File: .github/actions/isPullRequestMergeable/isPullRequestMergeable.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            owner: GITHUB_OWNER,
+            repo: EXPENSIFY_CASH_REPO,
+            pull_number: pullRequestNumber,
+        })
+            .then(({data}) => {
+                if (!_.isNull(data.mergeable)) {
+                    mergeabilityResolved = true;
+                    isMergeable = data.mergeable;
+                }
+            })
```
Comment (AndrewGable) [inline .github/actions/isPullRequestMergeable/isPullRequestMergeable.js:26]: Should we log here that we saw null? Also should we add a sleep and/or max loop param? I feel like we could end up hammering the API on accident.

File: .github/libs/promiseWhile.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+/**
+ * Simulates a do-while loop where the condition is determined by the result of a Promise.
+ *
+ * @param {Function} condition
+ * @param {Function} action
+ * @returns {Promise}
+ */
+function promiseDoWhile(condition, action) {
```
[the remark this answers is not in the document]
Comment (roryabraham) [reply .github/libs/promiseWhile.js:30]: No ... I could nix it if you want, but it does work! I saved this code from a while back when we were thinking about using it for a do-while loop.

File: .github/actions/isPullRequestMergeable/isPullRequestMergeable.js (hunk 1 above)
Comment (roryabraham) [reply .github/actions/isPullRequestMergeable/isPullRequestMergeable.js:26]: Sure, which do you think is better? Sleep or max loop? Both?

File: .github/libs/promiseWhile.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (AndrewGable) [reply .github/libs/promiseWhile.js:30]: In general let's not commit unused code and just add it when it's needed 👍

File: .github/actions/isPullRequestMergeable/isPullRequestMergeable.js (hunk 1 above)
Comment (AndrewGable) [reply .github/actions/isPullRequestMergeable/isPullRequestMergeable.js:26]: Maybe we should start out with a sane max default like: 10 second sleep, max 10 retries? What do you think?

Comment (roryabraham) [reply .github/actions/isPullRequestMergeable/isPullRequestMergeable.js:27]: Just added a `_.throttle` for now, along with a log that we saw null

Comment (roryabraham) [reply .github/actions/isPullRequestMergeable/isPullRequestMergeable.js:30]: I also added a `MAX_RETRIES` of 30 and throttle of 5 seconds. - 2 minutes 30 seconds total wait time before we give up.

File: .github/libs/promiseWhile.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (roryabraham) [reply .github/libs/promiseWhile.js]: Removed!

**Review by AndrewGable — APPROVED** [review]

**Review by Jag96 — APPROVED** [review]

**Pull request merged**